docs/langref/relay_expr.rst

```diff
@@ -0,0 +1,396 @@
+parameters, and a body expression (:py:class:`~tvm.relay.expr.Expr`)
+contained by curly braces.
+
+.. code-block:: python
+
+    def @id() { body }
+
+A definition may contain any number of parameters. For example, a
+simple function that invokes the `add` operator:
+
+.. code-block:: python
+
+    def @plus(%x, %y) { add(%x, %y) }
+
+Notice that within the function's body, the parameters are local
+variables, just like those bound in a :code:`let` expression.
+
+One may also annotate explicit types on definitions.
+For example, we can restrict the above definition to only work
+on certain types:
+
+.. code-block:: python
+
+    def @plus(%x: Tensor<(10, 10), float32>, %y: Tensor<(10, 10), float32>)
+               -> Tensor<(10, 10), float32> {
+        add(%x, %y)
+    }
+
+A function parameter is just a local variable (:py:class:`~tvm.relay.expr.LocalVar`) optionally
+annotated with a type. Parameters are written as :code:`%x : T`.
+
+When the type information is omitted, we will attempt to infer the most general type
+for the users. This property is known as generalization: for a definition without
+explicit annotations, we will attempt to assign the most general type. When the
+return type is omitted, we will infer the return type based on the function body.
+
+We can directly construct type-polymorphic definitions by writing down
+a set of type parameters for a definition. For example, one can define a
+polymorphic identity function for tensors as follows:
+
+.. code-block:: python
+    def @id<s: Shape, bt: BaseType>(%x: Tensor<s, bt>) {
+        %x
+    }
+
+Notice that the return type is omitted and will be inferred.
+
+A function may also be subject to one or more type relations, such as in
+the following:
+
+.. code-block:: python
+
+    def @plus(%x, %y) where Broadcast { add(%x, %y) }
+
+In the above definition, the types of `%x` and `%y` and the return type
+are subject to the `Broadcast` relation, meaning all three must be tensors
+and their shapes follow the elementwise broadcast relation. As with
+operators, the definitions of relations are not transparent to Relay
+and they are instead implemented externally in either C++ or Python.
+
+As in the case of `Broadcast`, relations are used to express complicated
+constraints on types (especially tensor shapes).
+All function relations must hold at all call sites;
+type checking is thus treated as a constraint-solving problem.
+For more detail on type relations and their implementations,
+please see the documentation on typing in Relay.
+
+See :py:class:`~tvm.relay.expr.Function` for the definition and documentation of function nodes.
+
+Operators
+=========
+
+An operator is a primitive operation not defined in the Relay
+language. Operators are declaredi n the global operator
+registry in C++. Many common operators are backed by TVM's
+Tensor Operator Inventory (`TOPI <https://github.com/dmlc/tvm/tree/master/topi>`__).
+
+but registered in the global environment in either Python
```

Comment:
This paragraph seems disconnected with the previous sentence. Reconnect?

Reply:
Thanks, will correct it